modules/core/src/main/java/com/google/refine/importers/ImporterUtilities.java

```diff
@@ -198,6 +198,26 @@ static public void setupColumns(Project project, List<String> columnNames) {
         project.columnModel.update();
     }
 
+    /**
+     * @param columnsHasData
+     *            Record if there is data in each column( false:null;true:has data)
+     */
```

Comment:
I think this needs an updated docstring now that it's moved.

Reply:
tbh the method in its current form is not very reusable imo. I kept it as is because it was already reviewed and as a first step my only goal was to fix the bugs by moving around the existing functionality. but the method is quite specific with its List of Booleans that need to be translated to the column indices. It would probably be easier to  understand and reuse this method if the input was a project and a List of Integers with columnIndices to delete. wdyt?

Reply:
update: i went ahead and implemented the suggested changes (https://github.com/OpenRefine/OpenRefine/pull/7272/commits/7005d8bfbb639042991195a3490108d30c034197) . I think they improve the code quite a bit, explanations in this review discussion https://github.com/OpenRefine/OpenRefine/pull/7272#discussion_r2104364044 

Let's continue the conversation in the linked thread, so reviewers don't need to jump between two threads, I will therefore mark this conversation as resolved, hope that is ok.